study-builder/src/main/java/org/broadinstitute/ddp/studybuilder/EmailBuilder.java

```diff
@@ -0,0 +1,229 @@
+        }
+        updateForConfigs(found);
+    }
+
+    private void updateForConfigs(List<Config> emailCfgs) {
+        List<EmailInfo> emails = new ArrayList<>();
+        for (int i = 0; i < emailCfgs.size(); i++) {
+            Config emailCfg = emailCfgs.get(i);
+            EmailInfo email = updateSendGridEmail(emailCfg);
+            if (email != null) {
+                emails.add(email);
+            }
+            if ((i + 1) % PROGRESS_STEP_COUNT == 0) {
+                LOG.info("Processed {}/{} emails", i + 1, emailCfgs.size());
+            }
+        }
+
+        if (!emails.isEmpty()) {
+            String table = emails.stream()
+                    .map(email -> String.format("name=%s templateId=%s versionId=%s",
+                            email.name, email.templateId, email.versionId))
+                    .collect(Collectors.joining("\n"));
+            LOG.info("Updated active versions of {} email templates:\n{}", emails.size(), table);
+        }
+    }
+
+    private EmailInfo updateSendGridEmail(Config emailCfg) {
+        File file = cfgPath.getParent().resolve(emailCfg.getString("filepath")).toFile();
+        String html = readRawHtmlContent(file);
+        if (emailCfg.getBoolean("render")) {
+            html = renderHtmlContent(file.getName(), html);
+        }
+        String subject = emailCfg.getString("subject");
+        String name = emailCfg.getString("name");
+        String templateId = emailCfg.getString("templateId");
+
+        var versionResult = sendGrid.getTemplateActiveVersionId(templateId);
+        versionResult.rethrowIfThrown(DDPException::new);
+        if (versionResult.hasError()) {
+            LOG.error("Could not find active version for email template id {}: {}",
+                    templateId, versionResult.getError());
+            return null;
+        }
+        String versionId = versionResult.getBody();
+
+        // Note: we're not updating version name since we didn't update template name, so we keep them the same.
+        var updateResult = sendGrid.updateTemplateVersion(templateId, versionId, null, subject, html);
+        updateResult.rethrowIfThrown(DDPException::new);
+        if (updateResult.hasError()) {
+            LOG.error("Error while updating active version {} for email template id {}: {}",
+                    versionId, templateId, versionResult.getError());
+            return null;
+        }
+
+        return new EmailInfo(name, templateId, versionId);
+    }
+
+    private String readRawHtmlContent(File file) {
+        if (!file.exists()) {
+            throw new DDPException("File " + file + " does not exist");
+        }
+        try {
+            return Files.readString(file.toPath());
+        } catch (IOException e) {
+            throw new DDPException(e);
+        }
+    }
+
+    private String renderHtmlContent(String name, String rawHtml) {
+        var ctx = new VelocityContext(varsCfg.root().unwrapped());
```

Comment:
Right now, we're just making the `varsCfg` object (which includes things from `vars.conf` and `subs.conf`) available for rendering. If needed, we can probably expand this to the `studyCfg` object, which includes all the study config values.